api/python/t4/packages.py

```diff
@@ -340,6 +346,25 @@ def __getitem__(self, prefix):
                 result.set(new_key, entry)
         return result
 
+    def fetch(self, dest):
+        """
+        Copy all descendants to dest. Descendants are written under their logical
+        names _relative_ to self. So if p[a] has two children, p[a][b] and p[a][c],
+        then p[a].fetch("mydir") will produce the following:
+            mydir
+                b
+                c
+
+        Args:
+            dest: where to put the files (locally)
+
+        Returns:
+            None
+        """
+        # TODO: do this with improved parallelism? connections etc. could be reused
+        for key, entry in self._data.items():
+            entry.fetch(os.path.join(dest, key))
```

Comment:
`dest` should probably be treated as a URL, not a path? That's how it works in `PackageEntry.fetch`